### PR TITLE
`Dictionaries Should Be Equal` added support for ignoring keys 

### PR DESCRIPTION
**Pull request by yuriverweij**
Issue #2717 
Added additional optional parameter to provide a list of keys to ignore in the comparison 

- [x] updated keyword documentation
- [x] added acceptance tests

### CODE REVIEW

File: src/robot/libraries/Collections.py [hunk 1]
```diff
@@ -737,19 +737,28 @@ def dictionary_should_not_contain_value(self, dictionary, value, msg=None):
...
+    def dictionaries_should_be_equal(self, dict1, dict2, msg=None, values=True, ignore_keys=None):
         """Fails if the given dictionaries are not equal.
 
         First the equality of dictionaries' keys is checked and after that all
         the key value pairs. If there are differences between the values, those
         are listed in the error message. The types of the dictionaries do not
         need to be same.
 
+        ``ignore_keys`` can be used to provide a list of keys to ignore in the
+        comparison.
```
Comment (pekkaklarck) [inline src/robot/libraries/Collections.py:749]: Need a note when this option has been added.

File: src/robot/libraries/Collections.py [hunk 2]
```diff
@@ -737,19 +737,28 @@ def dictionary_should_not_contain_value(self, dictionary, value, msg=None):
...
         See `Lists Should Be Equal` for more information about configuring
         the error message with ``msg`` and ``values`` arguments.
         """
+
+        if ignore_keys is None:
+            ignore_keys = []
         self._validate_dictionary(dict1)
         self._validate_dictionary(dict2, 2)
+        dict1 = {k: v for k, v in dict1.items() if k not in ignore_keys}
+        dict2 = {k: v for k, v in dict2.items() if k not in ignore_keys}
```
Comment (pekkaklarck) [inline src/robot/libraries/Collections.py:760]: This should be done only when `ignore_keys` is set. Now dictionaries are always recreated which wastes resources and in with custom dict subsclasses can also change semantics.

File: src/robot/libraries/Collections.py [hunk 3]
```diff
@@ -737,19 +737,28 @@ def dictionary_should_not_contain_value(self, dictionary, value, msg=None):
...
         are listed in the error message. The types of the dictionaries do not
         need to be same.
 
+        ``ignore_keys`` can be used to provide a list of keys to ignore in the
+        comparison.
+
         See `Lists Should Be Equal` for more information about configuring
         the error message with ``msg`` and ``values`` arguments.
         """
+
```
Comment (pekkaklarck) [inline src/robot/libraries/Collections.py:754]: We don't generally use empty lines inside functions or methods.

**Review by pekkaklarck — APPROVED** [review]
First of all, sorry for a very late review. I had missed this PR earlier and now noticed it when finalizing RF 6.1. This looks so good that I'll merge it and fix the small issues mentioned in separate comments.

I also plan to change this so that the list of ignored keys can be given as a Python list literal. That would be easiest to implement by having `list` as a type hint for `ignore_keys`, but because this library doesn't use type hints otherwise (it should!) I'll probably implement it using `ast.literal_eval` separately.

**Comment by yuriverweij** [comment]
No problem. Thanks for the feedback!

**Pull request merged**